astropy/io/fits/tests/test_hdulist.py

```diff
@@ -547,9 +547,8 @@ def test_new_hdulist_extend_keyword(self):
 
         h0 = fits.Header()
         hdu = fits.PrimaryHDU(header=h0)
-        sci = fits.ImageHDU(data=np.array(10))
-        image = fits.HDUList([hdu, sci])
-        image.writeto(self.temp("temp.fits"))
+        sci = fits.ImageHDU(data=np.array([10]))
+        hdus = fits.HDUList([hdu, sci])
```

Comment:
`image.writeto(self.temp("temp.fits"))` was removed from the test altogether. Is this intentional?

Reply:
Intentional. The file wasn't being read back and writing to a file shouldn't affect what this test is supposed to do (EXTEND should be added to the header as soon as it is inserted in HDUList). In other words, writing to a file wasn't doing anything here. But I can add it back and check the header after reading the file?

Reply:
Even if not read back it, it would still test that it does not break the writer. But how I would update this test, I am not sure. Should we keep the original test code but now slap a `with pytest.raises` on it instead? I'll let @saimn ponder.

Reply:
It's possible that at some point the behavior in PyFITS/Astropy was that the EXTEND keyword would only get added if the list was written to a file, which would explain the logic here. But the description of the test specifically mentions "adding a PrimaryHDU to a new HDUList object updates the EXTEND keyword on that HDU", so writing to a file should be needed. That said, it could make sense to write it to a file and then do the assertion on the header of file that is read back.

Reply:
I added back file writing (and now checking on read) on https://github.com/astropy/astropy/pull/14528/commits/83b6cfac94167ab24bd7e873b340869f8faf5452

Reply:
Yes probably in the past EXTEND was added only when writing the file. Anyway the best check is to write and read because what matters is to check that the file has been written correctly (which the test wasn't doing before :grimacing: )